spark-indexer/src/main/java/org/uniprot/store/spark/indexer/proteome/converter/DatasetProteomeEntryConverter.java

```diff
@@ -19,26 +30,338 @@
+        BookBuilder bookBuilder = new BookBuilder();
+        populateCommon(row, bookBuilder);
+        if (hasFieldName(FIRST, row)) {
+            bookBuilder.firstPage(String.valueOf(row.getLong(row.fieldIndex(FIRST))));
+        }
+        if (hasFieldName(LAST, row)) {
+            bookBuilder.lastPage(String.valueOf(row.getLong(row.fieldIndex(LAST))));
+        }
+        if (hasFieldName(VOLUME, row)) {
+            bookBuilder.volume(String.valueOf(row.getLong(row.fieldIndex(VOLUME))));
+        }
+        return bookBuilder.build();
+    }
+
+    private Citation getPatent(Row row) {
+        PatentBuilder patentBuilder = new PatentBuilder();
+        populateCommon(row, patentBuilder);
+        return patentBuilder.build();
+    }
+
+    private Citation getThesis(Row row) {
+        ThesisBuilder thesisBuilder = new ThesisBuilder();
+        populateCommon(row, thesisBuilder);
+        return thesisBuilder.build();
+    }
+
+    private Citation getSubmission(Row row) {
+        SubmissionBuilder submissionBuilder = new SubmissionBuilder();
+        populateCommon(row, submissionBuilder);
+        if (hasFieldName(DB, row)) {
+            submissionBuilder.submittedToDatabase(
+                    SubmissionDatabase.typeOf(row.getString(row.fieldIndex(DB))));
+        }
+        return submissionBuilder.build();
+    }
+
+    private Citation getJournalArticle(Row row) {
+        JournalArticleBuilder journalArticleBuilder = new JournalArticleBuilder();
+        populateCommon(row, journalArticleBuilder);
+        if (hasFieldName(FIRST, row)) {
+            journalArticleBuilder.firstPage(String.valueOf(row.getLong(row.fieldIndex(FIRST))));
+        }
+        if (hasFieldName(LAST, row)) {
+            journalArticleBuilder.lastPage(String.valueOf(row.getLong(row.fieldIndex(LAST))));
+        }
+        if (hasFieldName(VOLUME, row)) {
+            journalArticleBuilder.volume(String.valueOf(row.getLong(row.fieldIndex(VOLUME))));
+        }
+        if (hasFieldName(NAME, row)) {
+            journalArticleBuilder.journalName(row.getString(row.fieldIndex(NAME)));
+        }
+        return journalArticleBuilder.build();
+    }
+
+    private Citation getElectronicArticle(Row row) {
+        ElectronicArticleBuilder electronicArticleBuilder = new ElectronicArticleBuilder();
+        populateCommon(row, electronicArticleBuilder);
+        return electronicArticleBuilder.build();
+    }
+
+    private Citation getLiterature(Row row) {
+        LiteratureBuilder literatureBuilder = new LiteratureBuilder();
+        populateCommon(row, literatureBuilder);
+        return literatureBuilder.build();
+    }
+
+    private Citation getUnpublished(Row row) {
+        UnpublishedBuilder unpublishedBuilder = new UnpublishedBuilder();
+        populateCommon(row, unpublishedBuilder);
+        return unpublishedBuilder.build();
+    }
+
+    private void populateCommon(Row row, AbstractCitationBuilder citationBuilder) {
```

Comment:
Add annotation around method to avoid sonar warning
`@SuppressWarnings("squid:S3740")`